### PR TITLE
BUG: Avoid bare except clauses

### PR DESCRIPTION
**Pull request by eric-wieser**
Bare except is very rarely the right thing, as it swallows `SystemExit` and `KeyboardInterrupt`

`Exception` isn't much better, but fixing that requires more thought (and can always be done later)

This was a straight find and replace, but ignoring `tempita` since that's supposedly copied from elsewhere

### CODE REVIEW

**Comment by charris** [comment]
Tempita is vendorized, but also has some fixes above the original, so no problem making changes there. I did send some of the fixes upstream...

**Comment by eric-wieser** [comment]
Actually, I think the tempita ones should be bare, since they're seemingly just to add line number information

**Comment by charris** [comment]
LGTM, thanks Eric.

File: numpy/distutils/command/config.py [hunk 1]
```diff
@@ -441,7 +441,7 @@ def get_output(self, body, headers=None, include_dirs=None,
                 src, obj, exe = self._link(body, headers, include_dirs,
                                            libraries, library_dirs, lang)
                 grabber.restore()
-            except:
+            except Exception:
                 output = grabber.data
                 grabber.restore()
                 raise
```
Comment (njsmith) [inline numpy/distutils/command/config.py]: This one might have been correct as is... I'm not really sure what's going on here :-), but in general bare `except` followed by bare `raise` is an acceptable pattern.

**Review by njsmith — COMMENTED** [review]
Wow, I read through all these because I was nervous about blindly changing them all, but it turns out 99% of the cases really were obviously broken. Thanks for catching this.

File: numpy/distutils/command/config.py (hunk 1 above)
Comment (eric-wieser) [inline numpy/distutils/command/config.py]: That one looks like it should just be `try ... finally`

Comment (eric-wieser) [inline numpy/distutils/command/config.py]: Also, that function has been deprecated for almost 9 years anyway

**Pull request merged**